webapp/src/components/sidebar_right/sidebar_right.jsx

```diff
@@ -44,11 +45,57 @@ export default class SidebarRight extends React.PureComponent {
         yourAssignments: PropTypes.arrayOf(PropTypes.object),
         rhsState: PropTypes.string,
         theme: PropTypes.object.isRequired,
+        actions: PropTypes.shape({
+            getReviews: PropTypes.func.isRequired,
+            getUnreads: PropTypes.func.isRequired,
+            getYourPrs: PropTypes.func.isRequired,
+            getYourAssignments: PropTypes.func.isRequired,
+        }).isRequired,
     };
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            refreshing: false,
+        };
+    }
+
+    refresh = async (e) => {
+        if (this.state.refreshing) {
+            return;
+        }
+
+        if (e) {
+            e.preventDefault();
+        }
+
+        this.setState({refreshing: true});
+        let refreshAction;
+        switch (this.props.rhsState) {
+        case RHSStates.PRS:
+            refreshAction = this.props.actions.getYourPrs;
+            break;
+        case RHSStates.REVIEWS:
+            refreshAction = this.props.actions.getReviews;
+            break;
+        case RHSStates.UNREADS:
+            refreshAction = this.props.actions.getUnreads;
+            break;
+        case RHSStates.ASSIGNMENTS:
+            refreshAction = this.props.actions.getYourAssignments;
+            break;
+        default:
+            return;
+        }
+        await refreshAction();
```

Comment:
If `refreshAction` were to throw an exception, I believe we'd be stuck in `refreshing: true` permanently. Should we add a `try/catch` here?